gh-90437: Fix __main__.py documentation wording (#30480)
python/cpython: Doc/library/__main__.rst

    >>> asyncio.__main__.__name__
    'asyncio.__main__'

> This won’t work for __main__.py files in the root directory of a .zip file though.
> Hence, for consistency, minimal __main__.py without a __name__ check is preferred.

Reviewer: ```suggestion
This won’t work for ``__main__.py`` files in the root directory of a
``.zip`` file though.  Hence, for consistency, minimal ``__main__.py``
without a ``__name__`` check is preferred.
```

    >>> asyncio.__main__.__name__
    'asyncio.__main__'

> This won’t work for ``__main__.py`` files in the root directory of a

Reviewer 2: Was the change from a straight-up-and-down ASCII apostrophe (`won't`) to a "fancy" Unicode U+2019 RIGHT SINGLE QUOTATION MARK glyph (`won’t`) intentional here?

Reviewer 3: Please change this back to an apostrophe.

Reviewer 4: I think apostrophe is enough for consistency.

As I going through the python documentation, `won't` commonly appear as straight apostrophe.

Like:

- 1 `won't` appearance: https://docs.python.org/3/library/functions.html
- 2 `won't` appearances: https://docs.python.org/3/library/__main__.html

> This won’t work for ``__main__.py`` files in the root directory of a

Reviewer 2: A suggested change implementing the correction requested, so that anyone with commit access can apply it. (Assuming @riqts didn't disable maintainer commits.)

```suggestion
This won't work for ``__main__.py`` files in the root directory of a
```

> ``.zip`` file though.  Hence, for consistency, minimal ``__main__.py``

Reviewer 3: ISTM the only changes here should be:

    minimal -> a minimal
    are preferred -> is preferred

Reviewer 2: @rhettinger The primary motivator for this change was to eliminate the mention of "`venv` one mentioned above", as it is no longer mentioned above this paragraph. The new wording was hashed out at https://bugs.python.org/issue46279

Reviewer 2: (Er, which is also now #90437.)

Reviewer 4: To avoid backward reference of `venv`? I agree.

Reviewer 5: ```suggestion
``.zip`` file though.  Hence, for consistency, a minimal ``__main__.py``
```
